src/query/storage/promremote/storage.go

```diff
@@ -189,24 +186,25 @@ type promStorage struct {
 	client          *http.Client
 	endpointMetrics map[string]*instrument.HttpMetrics
 	scope           tally.Scope
-	samplesWritten  tally.Counter
-	droppedWrites   tally.Counter
-	enqueued        tally.Counter
-	enqueueErr      tally.Counter
-	dupWrites       tally.Counter
-	ingestorWrites  tally.Counter
-	batchWrite      tally.Counter
-	batchWriteErr   tally.Counter
-	tickWrite       tally.Counter
-	retryWrite      tally.Counter
-	logger          *zap.Logger
-	queryQueue      chan *storage.WriteQuery
-	workerPool      xsync.WorkerPool
-	pendingQuery    map[tenantKey]*WriteQueue
-	noTenantFound   tally.Counter
-	cancel          context.CancelFunc
-	writeLoopDone   chan struct{}
-	wrongTenant     tally.Counter
+	// Don't measure WriteQuery it is a very weird M3 internal data structure.
+	// samples are # of data points inside each WriteQuery
```

Comment:
what's weird about `WriteQuery`?

Reply:
it isn't consistent with the actual concept of neither time series nor samples, it is not # of remote write http requests as well, can't cross validate with Thanos metrics or even M3's own metrics